src/DotNetWorker.Core/FunctionMetadata/DefaultFunctionMetadata.cs

```diff
@@ -20,15 +36,69 @@ public class DefaultFunctionMetadata : IFunctionMetadata
         public bool ManagedDependencyEnabled { get; set; }
 
         /// <inheritdoc/>
-        public string? Name { get; set; }
+        public string? Name { get => _name; set => ClearIdAndSet(value, ref _name); }
 
         /// <inheritdoc/>
-        public string? EntryPoint { get; set; }
+        public string? EntryPoint { get => _entryPoint; set => ClearIdAndSet(value, ref _entryPoint); }
 
         /// <inheritdoc/>
         public IList<string>? RawBindings { get; set; }
 
         /// <inheritdoc/>
-        public string? ScriptFile { get; set; }
+        public string? ScriptFile { get => _scriptFile; set => ClearIdAndSet(value, ref _scriptFile); }
+
+        private static string? HashFunctionId(DefaultFunctionMetadata function)
```

Comment:
@jviau Could you add a link in the PR description to the host hash function this is adapted from? I think it would also be helpful to add a brief comment in this area mentioning that this is adapted from the host's hash function. 

If any other devs want to adapt our OOP handling of this later, we would have a clearer reverse engineering path they can follow to get the appropriate implementation context.